### PR TITLE
add 'list-freedomstudio-helpers' target

### PR DESCRIPTION
**Pull request by sifivekevin**
Adding manually managed meta information that Freedom Studio can use to ensure the best possible out-of-box experience when create new example software projects.  Discuss and review!

### CODE REVIEW

**Review by nategraff-sifive — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -192,10 +192,33 @@ clean:
...
+#
+# compiler_version: can be as general or specific as necessary
+# if omitted, then any or only compiler is used. If specified
+# compiler is not present then FS will display a message to
+# the user.
+#
+#############################################################
+.PHONY: list-freedomstudio-helpers
+list-freedomstudio-helpers:
+	@echo compiler=gcc, compiler_version=10
```
Comment (bsousi5) [inline Makefile:221]: In the comment, there is add_fcommon variable, that is not needed? If needed, what select true or false? is it compiler_version?

Comment (sifivekevin) [reply Makefile:220]: Just removed this.  Was thinking about adding that, but have decided not to. There are (or will be) better ways of handling this.

**Review by bsousi5 — APPROVED** [review]
Thinking more about it, these variables (compiler and compiler_version) may need to be a list. Unless for version it is imply that its 8.3 and 10.1. 
Will leave to software team to decide how to tackle this in the future, but IMO, it is better to be explicit something like = {gcc, llvm} and = {[8.3, 10.1], []}.

**Pull request merged**